### PR TITLE
clone handler: handle individual flags more explicitly

### PR DESCRIPTION
**Pull request by sporksmith**
This is mostly a refactor in preparation for supporting fork. https://github.com/shadow/shadow/issues/1987

There are also a couple minor behavioral changes, that shouldn't affect any of our supported thread libraries:

* Some flags that we don't actually care about are no longer required (e.g. CLONE_SYSVEM).
* CLONE_SETTLS *is* now required. Previously we would have tried to go ahead and execute the native clone without it, but the shim would misbehave since it depends on thread local storage.

### CODE REVIEW

File: src/main/host/syscall/handler/clone.rs [hunk 1]
```diff
@@ -64,55 +64,154 @@ impl SyscallHandler {
...
                 warn!("Exit signal is unimplemented");
                 return Err(Errno::ENOTSUP.into());
             }
+
+            assert!(!native_flags
+                .contains(CloneFlags::CLONE_CHILD_SETTID | CloneFlags::CLONE_CHILD_CLEARTID));
+            let ctid = ForeignPtr::<libc::pid_t>::null();
+
+            assert!(!native_flags.contains(CloneFlags::CLONE_PARENT_SETTID));
+            let ptid = ForeignPtr::<libc::pid_t>::null();
```
Comment (stevenengler) [inline src/main/host/syscall/handler/clone.rs:185]: Add comment explaining what these assertions are for?

File: src/main/host/syscall/handler/clone.rs [hunk 2]
```diff
@@ -64,55 +64,154 @@ impl SyscallHandler {
...
-                | CloneFlags::CLONE_CHILD_SETTID
-                | CloneFlags::CLONE_CHILD_CLEARTID;
-            let filtered_flags = flags.difference(flags_to_emulate);
             if raw_exit_signal != 0 {
                 warn!("Exit signal is unimplemented");
                 return Err(Errno::ENOTSUP.into());
             }
+
+            assert!(!native_flags
+                .contains(CloneFlags::CLONE_CHILD_SETTID | CloneFlags::CLONE_CHILD_CLEARTID));
```
Comment (stevenengler) [inline src/main/host/syscall/handler/clone.rs:181]: This only asserts that both flags aren't in `native_flags`, but it's still fine if just one of the flags is in `native_flags`. Is this intended?

**Review by stevenengler — APPROVED** [review]

File: src/main/host/syscall/handler/clone.rs (hunk 2 above)
Comment (sporksmith) [reply src/main/host/syscall/handler/clone.rs]: Oops, you're right this was a bug.

File: src/main/host/syscall/handler/clone.rs (hunk 1 above)
Comment (sporksmith) [reply src/main/host/syscall/handler/clone.rs]: Rearranged and added comments

**Pull request merged**